nrf_wifi/fw_if/umac_if/src/event.c

```diff
@@ -927,62 +927,93 @@ static enum nrf_wifi_status umac_process_sys_events(struct nrf_wifi_fmac_dev_ctx
 	case NRF_WIFI_EVENT_MODE_SET_DONE:
 		{
 			struct nrf_wifi_event_raw_config_mode *mode_event;
-
 			mode_event = (struct nrf_wifi_event_raw_config_mode *)sys_head;
+			struct tx_config *config = &def_dev_ctx->tx_config;
+			unsigned char if_idx = mode_event->if_index;
+			struct nrf_wifi_fmac_vif_ctx *vif = def_dev_ctx->vif_ctx[if_idx];
+
 			if (!mode_event->status) {
-				def_dev_ctx->vif_ctx[mode_event->if_index]->mode =
-								mode_event->op_mode;
-				/**
-				 * Set the transmit queue for RAW packet transmission
-				 */
+				vif->mode = mode_event->op_mode;
 #ifdef CONFIG_NRF700X_RAW_DATA_TX
-				if (mode_event->op_mode == (NRF_WIFI_TX_INJECTION_MODE |
-							    NRF_WIFI_STA_MODE)) {
-					def_dev_ctx->tx_config.peers[MAX_PEERS].peer_id =
-										MAX_PEERS;
-					def_dev_ctx->tx_config.peers[MAX_PEERS].if_idx =
-									mode_event->if_index;
-					def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
-									NRF_WIFI_STA_TX_INJECTOR;
-					def_dev_ctx->vif_ctx[mode_event->if_index]->txinjection_mode
-									= true;
-				} else
-#endif /* CONFIG_NRF700X_RAW_DATA_TX */
-				if (mode_event->op_mode == NRF_WIFI_STA_MODE) {
-					def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
-									NRF_WIFI_IFTYPE_STATION;
+				vif->txinjection_mode = false;
+#endif
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+				vif->promisc_mode = false;
+#endif
+				if ((mode_event->op_mode & NRF_WIFI_STA_MODE)
+								== NRF_WIFI_STA_MODE) {
+					mode_event->op_mode =
+						(mode_event->op_mode ^ NRF_WIFI_STA_MODE);
```

Comment:
Is this operation necessary? If yes can change this to `mode_event->op_mode ^= NRF_WIFI_STA_MODE;`

Reply:
This operation is to prevent going through each if check. If the mode is set to zero after the XOR operation, we can use that to prevent further `mode if checks`